### DISCUSSION TITLE
What is the best practice for integrating supabase google oauth login with Electron?

### DISCUSSION THREAD

**Discussion by pycraft114** [supabase/supabase#17722] [Questions]
```
<Button
        className={"min-w-[162px] min-h-[49px]"}
        onClick={async () => {
          await supabase.auth.signInWithOAuth({
            provider: "google",
          });
        }}
      >
```
I have a simple button which awaits authentication process via supabase signInWithOAuth.
And when user is successfully logged in it will redirect to a electron link (such as write://) which is set in the supabase dashboard.
some how after successful login supabase.auth.getUser() gives 401 error status code.

(Setting a redirect link to localhost or any other url beside file system directory wouldnt work since its a Electron app)

**Comment by igorlanko** [comment]
Have you tried making it `redirectTo` the URL schema of your app?

**Reply by nilekell** [reply]
@igorlanko When I attempt to use `redirectTo` with a custom protocol (see below code), I get the following error in the main process:

```
const { data, error } = await supabase.auth.signInWithOAuth({
        provider: 'google',
        options: {
        redirectTo: 'my-app://auth/callback',
        },
      })
```

```
(node:75182) electron: Failed to load URL: my-app://auth/callback?code=<auth-code> with error: ERR_UNEXPECTED
(Use `Electron --trace-warnings ...` to show where the warning was created)
```

Did you ever find a fix/workaround for this. Have been trying to get supabase OAuth to work with electron for a cople weeks now and it has been a struggle, any advice would be greatly appreciated.
